payroll/search.py

```diff
@@ -75,6 +73,33 @@ def _format_results(cls, results):
         return results
 
 
+class LazyPaginatedResults(list):
+    def __init__(self, *args):
+        results, hits = args
+
+        super().__init__(results)
+
+        self.results = results
+        self._hits = hits
+
+    def __getitem__(self, key):
+        if isinstance(key, slice):
+            # Because we're only returning pagesize results at a time, always
+            # return the slice from the start of the results.
+            # TO-DO: Be smarter about this. Slicing should be allowed within
+            # pagesize.
+            slice_len = key.stop - key.start
+            return [self.results[i] for i in range(0, slice_len, key.step or 1)]
+
+        return super().__getitem__(self, *args)
```

Comment:
hmm.. i sometimes you access the items through `self` and sometimes you access through `self.results` aren't those the same?